airflow/models/abstractoperator.py

```diff
@@ -154,37 +154,38 @@ def get_direct_relative_ids(self, upstream: bool = False) -> set[str]:
             return self.upstream_task_ids
         return self.downstream_task_ids
 
-    def get_flat_relative_ids(
-        self,
-        upstream: bool = False,
-        found_descendants: set[str] | None = None,
-    ) -> set[str]:
-        """Get a flat set of relative IDs, upstream or downstream."""
+    def get_flat_relative_ids(self, *, upstream: bool = False) -> set[str]:
+        """
+        Get a flat set of relative IDs, upstream or downstream.
+
+        Will recurse each relative found in the direction specified.
+
+        :param upstream: Whether to look for upstream or downstream relatives.
```

Comment:
Given we are adding the doctoring for it anyway, shall we add the line for type as well?

Reply:
hi @XD-DENG, i think that the types are now inferred from the annotations in the signature